### PR TITLE
Document options clause for remaning indexes and constraints

### PR DESCRIPTION
**Pull request by Hunterness**
Token lookup indexes and existence constraints now allow `OPTIONS` clause.

Also include other minor fixes/clean-up.

### CODE REVIEW

**Review by Lojjs — APPROVED** [review]

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 1]
```diff
@@ -24,10 +24,10 @@ This section comprises a glossary of all the keywords -- grouped by category and
...
+|<<administration-constraints-syntax-node-key, CREATE CONSTRAINT [node_key] [IF NOT EXISTS] FOR (n:Label) REQUIRE (n.prop1[, ..., n.propN]) IS NODE KEY [OPTIONS {optionKey: optionValue[, ...]}]>>  |  Schema | Create a constraint that ensures that all nodes with a particular label have all the specified properties and that the combination of property values is unique; i.e. ensures existence and uniqueness.
+|<<administration-constraints-syntax-rel-exists, CREATE CONSTRAINT [existence] [IF NOT EXISTS] FOR ()-"["r:REL_TYPE"]"-() REQUIRE r.property IS NOT NULL [OPTIONS {}]>>  | Schema   | Create a constraint ensuring that all relationships with a particular type have a certain property.
+|<<administration-constraints-syntax-create-unique, CREATE CONSTRAINT [uniqueness] [IF NOT EXISTS] FOR (n:Label) REQUIRE (n.prop1[, ..., n.propN]) IS UNIQUE [OPTIONS {optionKey: optionValue[, ...]}]>>  |  Schema | Create a constraint that ensures the uniqueness of the combination of node label and property values for a particular property key combination across all nodes.
```
Comment (Hunterness) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc:30]: This have NOT yet been built and checked so the links work

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 2]
```diff
@@ -75,6 +75,33 @@ Replaced by:
...
+label:added[]
+[source, cypher, role="noheader"]
+----
+CREATE CONSTRAINT [constraint_name] [IF NOT EXISTS]
+FOR ...
+REQUIRE ... IS NOT NULL
+OPTIONS "{" "}"
+----
+a|
+Existence constraints now allow an `OPTIONS` map, at this point there are no available values for the map.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
Existence constraints now allow an `OPTIONS` map, however, at this point there are no available values for the map.
```

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 3]
```diff
@@ -24,18 +24,18 @@ This section comprises a glossary of all the keywords -- grouped by category and
...
-|<<administration-constraints-syntax, CREATE CONSTRAINT [existence] [IF NOT EXISTS] FOR ()-"["r:REL_TYPE"]"-() REQUIRE r.property IS NOT NULL>>  | Schema   | Create a constraint ensuring that all relationships with a particular type have a certain property.
-|<<administration-constraints-syntax, CREATE CONSTRAINT [uniqueness] [IF NOT EXISTS] FOR (n:Label) REQUIRE (n.prop1[, ..., n.propN]) IS UNIQUE [OPTIONS {optionKey: optionValue[, ...]}]>>  |  Schema | Create a constraint that ensures the uniqueness of the combination of node label and property values for a particular property key combination across all nodes.
+|<<administration-constraints-syntax-node-exists, CREATE CONSTRAINT [existence] [IF NOT EXISTS] FOR (n:Label) REQUIRE n.property IS NOT NULL [OPTIONS {}]>>  | Schema   | Create a constraint ensuring that all nodes with a particular label have a certain property.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc]: Link doesn't work

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 4]
```diff
@@ -24,18 +24,18 @@ This section comprises a glossary of all the keywords -- grouped by category and
...
+|<<administration-constraints-syntax-node-key, CREATE CONSTRAINT [node_key] [IF NOT EXISTS] FOR (n:Label) REQUIRE (n.prop1[, ..., n.propN]) IS NODE KEY [OPTIONS {optionKey: optionValue[, ...]}]>>  |  Schema | Create a constraint that ensures that all nodes with a particular label have all the specified properties and that the combination of property values is unique; i.e. ensures existence and uniqueness.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc]: Link doesn't work

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 5]
```diff
@@ -24,18 +24,18 @@ This section comprises a glossary of all the keywords -- grouped by category and
...
+|<<administration-constraints-syntax-rel-exists, CREATE CONSTRAINT [existence] [IF NOT EXISTS] FOR ()-"["r:REL_TYPE"]"-() REQUIRE r.property IS NOT NULL [OPTIONS {}]>>  | Schema   | Create a constraint ensuring that all relationships with a particular type have a certain property.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc]: Link doesn't work

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc (hunk 1 above)
Comment (AlexicaWright) [reply cypher/cypher-docs/src/docs/dev/glossary.asciidoc:30]: Link doesn't work

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc (hunk 4 above)
Comment (AlexicaWright) [reply cypher/cypher-docs/src/docs/dev/glossary.asciidoc]:
```suggestion
|<<administration-constraints-syntax-node-key, CREATE CONSTRAINT [node_key] [IF NOT EXISTS] FOR (n:Label) REQUIRE (n.prop1[, ..., n.propN]) IS NODE KEY [OPTIONS {optionKey: optionValue[, ...]}]>>  |  Schema | Create a constraint that ensures all nodes with a particular label have all the specified properties and that the combination of property values is unique; i.e. ensures existence and uniqueness.
```

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc (hunk 5 above)
Comment (AlexicaWright) [reply cypher/cypher-docs/src/docs/dev/glossary.asciidoc]:
```suggestion
|<<administration-constraints-syntax-rel-exists, CREATE CONSTRAINT [existence] [IF NOT EXISTS] FOR ()-"["r:REL_TYPE"]"-() REQUIRE r.property IS NOT NULL [OPTIONS {}]>>  | Schema   | Create a constraint that ensures all relationships with a particular type have a certain property.
```

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc [hunk 6]
```diff
@@ -332,7 +336,8 @@ Listing indexes also allows for `WHERE` and `YIELD` clauses to filter the return
 
 [NOTE]
 ====
-The deprecated built-in procedures for listing indexes, such as `db.indexes`, work as before and are not affected by the <<administration-security-administration-database-indexes, `SHOW INDEXES` privilege>>.
+While the command for listing indexes require <<administration-security-administration-database-indexes, the `SHOW INDEX` privilege>>,
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]:
```suggestion
While the command for listing indexes require  the <<administration-security-administration-database-indexes, `SHOW INDEX` privilege>>,
```
But the link doesn't work

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc [hunk 7]
```diff
@@ -359,7 +364,7 @@ include::../indexes/drop-a-non-existing-index.asciidoc[leveloffset=+1]
 
 [NOTE]
 ====
-This syntax does not support dropping relationship property indexes, these can only be dropped by name.
+This syntax only support dropping node property indexes, all others can only be dropped by name.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]:
```suggestion
This syntax only supports dropping node property indexes, all others can only be dropped by name.
```

**Review by AlexicaWright — COMMENTED** [review]
Some links aren't working and some editorial suggestions.

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc (hunk 6 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]: I didn't even change the link? so it should work (or it's broken in general)

**Comment by Hunterness** [comment]
[Document remaining OPTIONS maps](https://trello.com/c/UMKkjpkM/2595-document-remaining-options-maps)

**Comment by AlexicaWright** [comment]
All links work now 👍 !

**Pull request merged**